crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -732,8 +732,13 @@ pub mod common_conditions {
     /// app.run(&mut world);
     /// assert_eq!(world.resource::<Counter>().0, 0);
     /// ```
-    pub fn in_state<S: States>(state: S) -> impl FnMut(Res<State<S>>) -> bool + Clone {
-        move |current_state: Res<State<S>>| *current_state == state
+    pub fn in_state<S: States, M: StateMatcher<S>>(
+        state: M,
+    ) -> impl FnMut(Option<Res<State<S>>>) -> bool + Clone {
+        move |current_state: Option<Res<State<S>>>| match current_state {
```

Comment:
Taking an Option here is a clever way to increase robustness: we should do this regardless.